### PR TITLE
added third party integration for provisioner framework

### PR DESCRIPTION
**Pull request by tanujashinde0405**
Signed-off-by: Tanuja Shinde <tanuja.shinde@seagate.com>

# Problem Statement
- added third party integration for provisioner framework

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: test/deploy/kubernetes/solution-config/config.yaml [hunk 1]
```diff
@@ -2,24 +2,21 @@ cortx:
...
-        - oldap-server2.cortx-cluster.com
-        - oldap-server3.cortx-cluster.com
+        - openldap-svc.default.svc.cluster.local
       admin: admin
       secret: openldap_admin_secret
       base_dn: dc=seagate,dc=com
     consul:
       endpoints:
-        - http://consul-server.cortx-cluster.com:80
+        - tcp://consulserver.default.svc.cluster.local:8301
```
Comment (nitin-seagate) [inline test/deploy/kubernetes/solution-config/config.yaml:19]: Change this to:
`tcp://consul-server.default.svc.cluster.local:8301`

File: test/deploy/kubernetes/solution-config/secrets.yaml [hunk 2]
```diff
@@ -4,10 +4,11 @@ metadata:
  name: cortx-secret
 type: Opaque
 stringData:
-  openldap_admin_secret: openldap_password
+  openldap_admin_secret: seagate2
   kafka_admin_secret: kafka_password
```
Comment (nitin-seagate) [inline test/deploy/kubernetes/solution-config/secrets.yaml:8]: Lets put default password for consul and kafka as:
`Seagate@123`

File: test/deploy/kubernetes/provisioner-pods/deployment_control_node.yaml [hunk 3]
```diff
@@ -44,7 +44,7 @@ spec:
     - /bin/sh
     - -c
     - set -x;
-      /opt/seagate/cortx/provisioner/bin/cortx_deploy -n $NODE_ID -f /etc/cortx/solution -c yaml:///etc/cortx/cluster.conf;
+      /opt/seagate/cortx/provisioner/bin/cortx_deploy -n $NODE_ID -f /etc/cortx/solution -c yaml:///etc/cortx/cluster.conf -m;
```
Comment (nitin-seagate) [inline test/deploy/kubernetes/provisioner-pods/deployment_control_node.yaml:47]: Remove mocking

File: test/deploy/kubernetes/provisioner-pods/deployment_storage_node1.yaml [hunk 4]
```diff
@@ -68,7 +68,7 @@ spec:
     - /bin/sh
     - -c
     - set -x;
-      /opt/seagate/cortx/provisioner/bin/cortx_deploy -n $NODE_ID -f /etc/cortx/solution -c yaml:///etc/cortx/cluster.conf;
+      /opt/seagate/cortx/provisioner/bin/cortx_deploy -n $NODE_ID -f /etc/cortx/solution -c yaml:///etc/cortx/cluster.conf -m;
```
Comment (nitin-seagate) [inline test/deploy/kubernetes/provisioner-pods/deployment_storage_node1.yaml:71]: Remove mocking

File: test/deploy/kubernetes/provisioner-pods/deployment_storage_node2.yaml [hunk 5]
```diff
@@ -68,7 +68,7 @@ spec:
     - /bin/sh
     - -c
     - set -x;
-      /opt/seagate/cortx/provisioner/bin/cortx_deploy -n $NODE_ID -f /etc/cortx/solution -c yaml:///etc/cortx/cluster.conf;
+      /opt/seagate/cortx/provisioner/bin/cortx_deploy -n $NODE_ID -f /etc/cortx/solution -c yaml:///etc/cortx/cluster.conf -m;
```
Comment (nitin-seagate) [inline test/deploy/kubernetes/provisioner-pods/deployment_storage_node2.yaml:71]: Remove mocking

File: test/deploy/kubernetes/provisioner-pods/deployment_storage_node3.yaml [hunk 6]
```diff
@@ -68,7 +68,7 @@ spec:
     - /bin/sh
     - -c
     - set -x;
-      /opt/seagate/cortx/provisioner/bin/cortx_deploy -n $NODE_ID -f /etc/cortx/solution -c yaml:///etc/cortx/cluster.conf;
+      /opt/seagate/cortx/provisioner/bin/cortx_deploy -n $NODE_ID -f /etc/cortx/solution -c yaml:///etc/cortx/cluster.conf -m;
```
Comment (nitin-seagate) [inline test/deploy/kubernetes/provisioner-pods/deployment_storage_node3.yaml:71]: Remove mocking

**Review by nitin-seagate — CHANGES_REQUESTED** [review]

**Review by nitin-seagate — APPROVED** [review]
LGTM

**Pull request merged**